### ISSUE TITLE
Difference between PyCurrentMover and c_GridCurrent Mover

### ISSUE THREAD

**Issue by tommasogallingani** [NOAA-ORR-ERD/PyGnome#142]
Hi, 
I was wondering which are the differences between PyCurrentMover and c_GridCurrentMover and why we are obtaining different results using the netcdf file attached. The grid file .nc attached is a "zero depth" file.
We have seen that using the c_GridCurrentMover there's almost no effect on particle moving while using the PyCurrentMover the effect is significant,
Thanks in advance
Tommaso

https://drive.google.com/file/d/1pDCuQF76hUFiNPhOz06a3ueaKj-wP6tx/view?usp=sharing

**Comment by ChrisBarker-NOAA** [comment]
The `c_GridCurrentMover` is the older C++ cpde that used in GNOME 1 (Desktop GNOME)

`PyCurrentMover` (which has been renamed to just "Current Mover" is newer code. The new version does a 2nd order Runge Kutta rather than Euler method and may have slightly better interpolation. 

They should give similar, though not identical results.

> c_GridCurrentMover there's almost no effect on particle moving while using the PyCurrentMover the effect is significant,

Are the PyCurrentMover results what you expect? If so, then the data file is probably somehow incompatible with the older code. There's no reason to use the older code [^], so I wouldn't worry about it :-)

If the results are not what you expect than you might have found a bug -- we'l take a look.

[*] There are some older, non-standards-compliant model output files that only work with the old code, which is why we keep it around (but have recently renamed things to make it a bit more hidden)

**Comment by ChrisBarker-NOAA** [comment]
Hmm -- at a quick glance, this is pretty simple file -- it should work with both.

@coconnor8: Do you see any reason this might not work with the C++ code ?

**Comment by coconnor8** [comment]
I was able to load the file into WebGnome, both as a c_GridCurrent and as a PyCurrentMover. They both moved a spill similarly, and the vectors looked close though not exactly the same. I will try it also just with PyGnome and see if there is a difference there.

**Comment by coconnor8** [comment]
Do you have a script you could post? I am not seeing much difference, but there could be something unique in your scenario.

**Comment by tommasogallingani** [comment]
Hi @ChrisBarker-NOAA and @coconnor8. We have further investigated the behaviour of the two movers and we found something that could maybe explaine the differences we have seen. In particular, we saw that for spill near the coast, where 3D Random mover is used it happens that the particles spreading is small. 

The situation is the following:
- we have surface spill near the coast where the currents value is limited for all depth
- the 3d mover is used to simulate limited vertical mixing
- the wind mover should be the major driving force here, right? But we saw in the code that only particles at surface (depth==0.0) are moved by wind mover. But of course, only in the first step of simulation particles are placed exactly at 0.0. Is that correct that there should be a limited layer where particles are affected by wind? In particular if particles are a proxy of a control volume with also surface oil concentration that is not zero.

does it make sense?

thanks

tommaso 


-

**Comment by ChrisBarker-NOAA** [comment]
yes, the wind mover acts only on particles exactly on the surface.

The idea is that it captures the direct wind effect on the surface sheen.

The assumption is that the the near-surface wind driven currents should be captured by the ocean circulation model.

If you wanted to capture the movement in the near-surface, I suppose we could have a wind-driven surface profile -- but in fact, the circulation in the region is extremely complex -- pretty much impossible to estimate it, so we make the clear distinction between "on the surface" and "under the surface".

But if you are trying to simulate a surface spill, then you really want to have your release on the surface, and it should stay there. Currently we don't have a mechanism to move particles off the surface.

The RandomMover3D simulated vertical diffusion within the water column -- it should be used for subsurface releases -- for non-buoyand substances or for oil with a RiseVelocity Mover.

**Comment by tommasogallingani** [comment]
thanks @ChrisBarker-NOAA. If I get you right, there's no way to simulate the vertical displacement due to wave turbolent mixing.
